app/src/main/java/io/gnosis/safe/ui/transaction/TransactionsViewModel.kt

```diff
@@ -15,29 +19,46 @@ class TransactionsViewModel
 ) : BaseStateViewModel<TransactionsViewState>(appDispatchers) {
 
     override fun initialState(): TransactionsViewState = TransactionsViewState(null, true)
-
     fun load() {
         safeLaunch {
             val safeAddress = safeRepository.getActiveSafe()!!.address
             val transactions = transactionRepository.getTransactions(safeAddress)
             updateState {
+                val newTransactions = newTransactions(transactions, safeAddress)
+                val transactionsWithSectionHeaders = addSectionHeaders(newTransactions)
```

Comment:
There is probably a more idiomatic way in Kotlin...

Reply:
for me naming of `newTransactions` is confusing. Maybe renaming it to mapTransactions or something similar will make it more clear